File: 수빈/src/PGS_Lv2_42842.java
public class PGS_Lv2_42842 {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int sum = brown + yellow;

        for (int i = 1; i * i <= sum; i++) {
            if (sum % i == 0 && (i - 2) * (sum / i - 2) == yellow) {
                answer[0] = sum / i;
                answer[1] = i;
            }
        }

        return answer;
    }
}

File: 성규/6주차/카펫/PGS_Lv2_42842.js
function solution(brown, yellow) {
  let sum = brown + yellow;

  for (let height = 3; height <= brown; height++) {
    if (sum % height === 0) {
      let weight = sum / height;
      if ((height - 2) * (weight - 2) === yellow) return [weight, height];
    }
  }
}
